apps/web/src/components/costs/energy-costs-last-seven-days-card.tsx

```diff
@@ -1,11 +1,56 @@
-import { Card, CardContent, CardHeader, CardTitle } from "@energyleaf/ui/card";
+import { calculateCosts } from "@/components/dashboard/energy-cost";
+import { getSession } from "@/lib/auth/auth.server";
+import { getElectricitySensorIdForUser, getEnergyDataForSensor } from "@/query/energy";
+import { getUserDataHistory } from "@/query/user";
+import { formatNumber } from "@energyleaf/lib";
+import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@energyleaf/ui/card";
+import { redirect } from "next/navigation";
+
+export default async function EnergyCostsLastSevenDays() {
+    const { session, user } = await getSession();
+
+    if (!session) {
+        redirect("/");
+        return;
+    }
+
+    const userId = user.id;
+    const sensorId = await getElectricitySensorIdForUser(userId);
+
+    if (!sensorId) {
+        return (
```

Comment:
Wird es auf der Kosten Seite auch Kacheln geben, wo Daten unabhängig vom Sensor angezeigt werden? Wenn ja, wäre vielleicht eine vorgelagerte Prüfung auf die Sensor-ID gut. Sonst hat man ja etliche Kacheln mit dem Hinweis "Ihr Sensor konnte nicht gefunden werden."

Reply:
Jo hast Recht. Hat Lennard hinzugefügt und habe seine Logik da jetzt übernommen in page.tsx